goal_src/jak2/levels/common/airlock.gc

```diff
@@ -7,3 +7,1762 @@
+               (ja :group! (get-art-by-name (-> self draw art-group) "idle" art-joint-anim)
+                   :num! (identity
+                     (the float (+ (-> (get-art-by-name (-> self draw art-group) "idle" art-joint-anim) frames num-frames) -1))
+                     )
+                   )
+               )
+           (and (-> self next-state) (= (-> self next-state name) 'close))
+           )
+          ((= v1-0 'front)
+           (let ((f30-0 (com-airlock-method-25 self (target-pos 0) #f))
+                 (f0-3 (com-airlock-method-25 self (camera-pos) #f))
+                 )
+             (and (< 2048.0 f30-0) (>= (* f30-0 f0-3) 0.0))
+             )
+           )
+          ((= v1-0 'back)
+           (let ((f30-1 (com-airlock-method-25 self (target-pos 0) #f))
+                 (f0-5 (com-airlock-method-25 self (camera-pos) #f))
+                 )
+             (and (< f30-1 -2048.0) (>= (* f30-1 f0-5) 0.0))
+             )
+           )
+          ((= v1-0 'sound)
+           (if (>= (com-airlock-method-25 self (target-pos 0) #f) 0.0)
+               (com-airlock-method-27 self (the-as symbol (-> arg3 param 0)))
+               )
+           )
+          ((= v1-0 'distance)
+           (* (the int (com-airlock-method-25 self (target-pos 0) #f)) 8)
+           )
+          )
+        )
+      )
+    )
+  :enter (behavior ((arg0 symbol))
+    (set! (-> self visible-move?) #f)
+    (none)
+    )
+  :exit (behavior ()
+    (when (nonzero? (-> self sound-id))
+      (let ((v1-2 (the-as sound-rpc-set-param (get-sound-buffer-entry))))
+        (set! (-> v1-2 command) (sound-command set-param))
+        (set! (-> v1-2 id) (-> self sound-id))
+        (set! (-> v1-2 params volume) -4)
+        (set! (-> v1-2 auto-time) 24)
+        (set! (-> v1-2 auto-from) 2)
+        (set! (-> v1-2 params mask) (the-as uint 17))
+        (-> v1-2 id)
+        )
+      (set! (-> self sound-id) (new 'static 'sound-id))
+      0
+      )
+    (when (nonzero? (-> self gear-sound-id))
+      (let ((v1-6 (the-as sound-rpc-set-param (get-sound-buffer-entry))))
+        (set! (-> v1-6 command) (sound-command set-param))
+        (set! (-> v1-6 id) (-> self gear-sound-id))
+        (set! (-> v1-6 params volume) -4)
+        (set! (-> v1-6 auto-time) 24)
+        (set! (-> v1-6 auto-from) 2)
+        (set! (-> v1-6 params mask) (the-as uint 17))
+        (-> v1-6 id)
+        )
+      (set! (-> self gear-sound-id) (new 'static 'sound-id))
+      0
+      )
+    (none)
+    )
+  :trans (behavior ()
+    (if (logtest? (-> self draw status) (draw-control-status on-screen))
+        (set! (-> self visible-move?) #t)
+        )
+    (when (and (open-door? self)
+               (begin
+                 (let ((gp-0 (res-lump-struct (-> self entity) 'on-activate structure)))
+                   (if gp-0
+                       (script-context-method-9
```

Comment:
at this point, I think it's probably safe to guess that `script-context-method-9` is `eval-script!` or something like that.